gstd/src/msg/macros.rs

```diff
@@ -64,6 +64,44 @@ macro_rules! impl_futures {
 
                 Ok(self)
             }
+
+            /// Execute a function when the reply is received.
+            ///
+            /// Note: This callback will run in reply context and consume reply gas.
+            ///
+            /// # Examples
+            ///
+            /// Send message to echo program and wait for reply.
+            ///
+            /// ```
+            /// use gstd::{ActorId, msg, debug};
+            ///
+            /// #[gstd::async_main]
+            /// async fn main() {
+            ///     let dest = ActorId::from(1); // Replace with correct actor id
+            ///
+            ///     msg::send_bytes_for_reply(dest, b"PING", 0, 0)
+            ///         .expect("Unable to send")
+            ///         .handle_reply(|| {
+            ///             debug!("reply code: {:?}", msg::reply_code());
+            ///
+            ///             if msg::load_bytes().unwrap_or_default() == b"PONG" {
+            ///                debug!("successfully received pong");
+            ///             }
+            ///         })
+            ///         .await
+            ///         .expect("Received error");
+            /// }
+            /// # fn main() {}
+            /// ```
+            ///
+            /// # Panics
+            ///
+            /// Panics if this is called second time.
+            pub fn handle_reply<F: FnMut() + 'static>(self, f: F) -> Self {
```

Comment:
Seems static lifetime might be a problem if I start using it within Sails service which manages its data as an external reference passed to its constructor:
https://github.com/gear-tech/sails/blob/f8f4d160ef20fa746e6e5c28701463c28e765bf1/examples/demo/app/src/counter/mod.rs#L26

Reply:
Take my words back. Apparently, if you implement the main idea of these functionality, I won't be able to do anything with non-static data as my service will be dead by the time of this callback is being called